app/grandchallenge/evaluation/views/__init__.py

```diff
@@ -980,3 +987,65 @@ def get_success_url(self):
 
     def get_permission_object(self):
         return self.get_object().challenge
+
+
+class ConfigureAlgorithmPhasesView(PermissionRequiredMixin, FormView):
+    form_class = ConfigureAlgorithmPhasesForm
+    permission_required = "evaluation.configure_algorithm_phase"
+    template_name = "evaluation/configure_algorithm_phases_form.html"
+    raise_exception = True
+
+    def form_valid(self, form):
+        for phase in form.cleaned_data["phases"]:
+            self.turn_phase_into_algorithm_phase(
+                phase=phase,
+                inputs=form.cleaned_data["algorithm_inputs"],
+                outputs=form.cleaned_data["algorithm_outputs"],
+            )
+        messages.success(self.request, "Phases were successfully updated")
+        return super().form_valid(form)
+
+    def get_success_url(self):
+        return reverse("challenges:list")
+
+    def turn_phase_into_algorithm_phase(self, *, phase, inputs, outputs):
+        archive = Archive.objects.create(
+            title=format_html(
+                "{challenge_name} {phase_title} dataset",
+                challenge_name=phase.challenge.short_name,
+                phase_title=phase.title,
+            ),
+            workstation=Workstation.objects.get(
+                slug=settings.DEFAULT_WORKSTATION_SLUG
+            ),
+            logo=(
+                phase.challenge.logo
+                if phase.challenge.logo
+                else create_uploaded_image()
+            ),
+        )
+        archive.full_clean()
+        archive.save()
+
+        for user in phase.challenge.admins_group.user_set.all():
+            archive.add_editor(user)
+
+        try:
+            challenge_request = ChallengeRequest.objects.get(
+                short_name=phase.challenge.short_name
+            )
+            phase.algorithm_time_limit = (
+                challenge_request.inference_time_limit_in_minutes
+            )
+        except ObjectDoesNotExist:
+            messages.warning(
+                self.request,
+                f"Algorithm run time limit could not be configured for {phase}.",
+            )
```

Comment:
It might be better to add this to the form and use the value from the challenge request as the initial value if it exists, then you avoid a lookup in `form_valid`.

Reply:
Ok, then we would presume that there can only be one time limit, so we can only configure phases from one challenge at a time.

Reply:
Ignore my last comment. I hadn't seen your earlier comment.